### DISCUSSION TITLE
How to load default settings in the migrations?

### DISCUSSION THREAD

**Discussion by denisgurev** [pocketbase/pocketbase#1191] [Q&A] [answered]
I've already migrated to v0.9 and everything works fine with the generated migrations when I'm making changes but I'm not sure how to load default app settings. From the release notes:

> This allow writting custom migration scripts such as programmatically creating collections,
initializing default settings, running data imports, etc., with a JavaScript API very similar to the Go one (more documentation will be available soon).

I guess there are no automigration for the settings but it should be possible to set default settings, right?

My use case is that I want to set predefined mail templates when users clone my app.

If possible, @ganigeorgiev could you add an example how to do it?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no automigration for the settings because most of the app configurations are environment specific.

Yes, you can write custom migrations that can preset the mail templates. As mentioned in the release notes, this currently is not documented since I'm not sure how to structure it yet, but here is an example how you could do it:

1. Run `./pocketbase migrate create 'set_email_templates'`
   _The above command will create a blank migration JS template with the provided name._

2. Navigate to the created migration and add the following content:
    ```js
    migrate((db) => {
      const dao = new Dao(db);
      const settings = dao.findSettings();

      settings.meta.verificationTemplate.subject = 'your default subject';
      settings.meta.verificationTemplate.body = '<p>Hello world - {ACTION_URL}!</p>';
      settings.meta.verificationTemplate.actionUrl = '{APP_URL}/_/#/auth/confirm-verification/{TOKEN}';

      // you can also change similarly any other template or setting
      // 
      // all setting fields are identical to the json response so you can
      // use the https://pocketbase.io/docs/api-settings/ > Update settings
      // "Body parameters" section as a reference

      return dao.saveSettings(settings);
    }, (db) => {
      // optionally you can also revert the above changes to their
      // original state when "migrate down" is executed...
    })
    ```

In general, you can use all `db` and `dao` helpers from the Go docs https://pocketbase.io/docs/database/, https://pocketbase.io/docs/record-methods/, https://pocketbase.io/docs/collection-methods/, etc. **with the only difference that the method name and fields are in camelCase** (if the field name is all CAPS -> then it is converted to all lowercase, eg. `c.JSON()` -> `c.json()`).

Proper documentation (_and eventually support to bind to the app hooks with js too_) will be available soon.

**Reply by ganigeorgiev** [reply]
Note that the migration will be executed automatically on `./pocketbase serve` and it will be executed only once, so any change after the first run will have no effect.

You can revert the last `n` migrations and reapply them by running `./pocketbase migrate down [n]` (`n` by default is 1) and then restart the server (or run `./pocketbase migrate up`).

**Reply by denisgurev** [reply]
Thank you, it works!

One last question, to check if I understand the settings docs, if I want to disable the logs in a migration the valid path would be `settings.logs.maxDays = 0`, right?

**Reply by ganigeorgiev** [reply]
Yes, that's correct.
